### PR TITLE
tiup: add document of tispark support in tiup cluster

### PR DESCRIPTION
**Pull request by AstroProfundis**
### What is changed, added or deleted? (Required)

Add documentations of (experimental) support of TiSpark in TiUP Cluster. It should be merged **after** https://github.com/pingcap/tiup/pull/531 is released.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### CODE REVIEW

File: tispark-deployment-topology.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+> **注意：**
+>
+> - 无需手动创建配置文件中的 `tidb` 用户，TiUP cluster 组件会在目标主机上自动创建该用户。可以自定义用户，也可以和中控机的用户保持一致。
+> - 如果部署目录配置为相对路径，会部署在用户家目录下。
+
+# 环境要求
+
+由于 TiSpark 基于 Apache Spark 集群，在启动包含 TiSpark 组件的 TiDB 集群前，需要在部署了 TiSpark 组件的服务器上安装 Java 运行时环境(JRE) 8, 否则将无法启动相关组件。
+
+TiUP 不提供自动安装 JRE 的支持，该操作需要用户自行完成。JRE 8 的安装方法可以参考 [OpenJDK 的文档说明](https://openjdk.java.net/install/)。
```
Comment (lonng) [inline tispark-deployment-topology.md:43]: 最好是 link 一下 TiSpark 的文档，用户安装之后怎么验证 TiSpark 是否启动成功。

Comment (AstroProfundis) [reply tispark-deployment-topology.md:43]: 在上面已经有了

File: tispark-deployment-topology.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
```
Comment (yikeke) [inline tispark-deployment-topology.md:1]: Please add this file to TOC.md. @AstroProfundis

File: tispark-deployment-topology.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+title: TiSpark 部署拓扑
+summary: 介绍 TiUP 部署包含 TiSpark 组件的 TiDB 集群的拓扑结构。
+category: how-to
+aliases: ['/docs-cn/dev/reference/tools/tispark/deploy/','/docs-cn/dev/tispark/deploy-tispark/']
```
Comment (yikeke) [inline tispark-deployment-topology.md:5]:
```suggestion
```
no alias needed for a new file

File: tispark-deployment-topology.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+---
+title: TiSpark 部署拓扑
+summary: 介绍 TiUP 部署包含 TiSpark 组件的 TiDB 集群的拓扑结构。
+category: how-to
```
Comment (yikeke) [inline tispark-deployment-topology.md:4]:
```suggestion
```

File: tiup/tiup-cluster.md [hunk 5]
```diff
@@ -407,6 +407,10 @@ tiup cluster patch test-cluster /tmp/tidb-hotfix.tar.gz -N 172.16.4.5:4000
 
 ## 导入 TiDB Ansible 集群
 
+> **注意：**
+>
+> TiUP Cluster 对 TiSpark 的支持目前为实验性特性，暂不支持导入启用了 TiSpark 组件的集群。
```
Comment (yikeke) [inline tiup/tiup-cluster.md:412]:
```suggestion
> TiUP cluster 组件对 TiSpark 的支持目前为实验性特性，暂不支持导入启用了 TiSpark 组件的集群。
```

File: production-deployment-using-tiup.md [hunk 6]
```diff
@@ -93,6 +93,10 @@ aliases: ['/docs-cn/dev/how-to/deploy/orchestrated/tiup/','/docs-cn/dev/tiflash/
 
     包含最小拓扑的基础上，同时部署 TiDB Binlog。TiDB Binlog 是目前广泛使用的增量同步组件，可提供准实时备份和同步功能。
 
+- [增加 TiSpark 拓扑架构](/tispark-deployment-topology.md)
+
+    包含最小拓扑的基础上，同时部署 TiSpark 组件。TiSpark 是 PingCAP 为解决用户复杂 OLAP 需求而推出的产品。TiUP Cluster 的 TiSpark 支持目前为实验性特性。
```
Comment (yikeke) [inline production-deployment-using-tiup.md:98]:
```suggestion
    包含最小拓扑的基础上，同时部署 TiSpark 组件。TiSpark 是 PingCAP 为解决用户复杂 OLAP 需求而推出的产品。TiUP cluster 组件对 TiSpark 的支持目前为实验性特性。
```

**Pull request merged**